Document hotfix process (#701)
openstack-k8s-operators/dataplane-operator: docs/assemblies/proc_hotfixing-the-data-plane-container-content-images.adoc

[id="proc_hotfixing-the-data-plane-container-content-images-{context}"]
= Hotfixing the data plane container content with images

[role="_abstract"]

> When container hotfix content is delivered as images, the container processes need to be restarted to use the new images. This will be accomplished by creating a new `OpenStackDataPlaneDeployment`.

Reviewer: ```suggestion
When container hotfix content is delivered as images, the container processes need to be restarted to use the new images. Create a new `OpenStackDataPlaneDeployment` custom resource (CR) that restarts the containers with the new images.
```
Active voice + removed future tense.

Author: this reads to me like the user needs to do something special or different in order create the deployment "that restarts...". Is there a way to say it so that the meaning is "Create a new OpenStackDataPlaneDeployment ... which will result in the containers being restarted"?

Reviewer 2: `This restarts the containers with the new images`

Author: this was resolved.

.Procedure

> . Optional: Prepare the container hotfix image in a container registry where the image can be pulled by affected data plane nodes

Reviewer: ```suggestion
. Optional: Prepare the container hotfix image in a container registry where the image can be pulled by affected data plane nodes:
```

+
----
$ podman pull <container_registry>/<container_project>/<container_image>:<container_tag>
$ podman tag <container_registry>/<container_project>/<container_image>:<container_tag> <updated_container_registry>/<updated_container_project>/<container_image>:<container_tag>
$ podman push <updated_container_registry>/<updated_container_project>/<container_image>:<container_tag>
----
+

> * Replace `<container_registry>` with the source registry for the already hotfixed container image
> * Replace `<container_project>` with the source project for the already hotfixed container image
> * Replace `<container_image>` with the already hotfixed container image
> * Replace `<container_tag>` with the tag being for the already hotfixed container image
> * Replace `<updated_container_registry>` with a container registry to serve the hotfixed container image. The OCP internal container image registry can be used.
> * Replace `<updated_container_project>` with a container project to use for the hotfixed container image.

Reviewer: ```suggestion
* Replace `<container_registry>` with the source registry for the hotfixed container image.
* Replace `<container_project>` with the source project for the hotfixed container image.
* Replace `<container_image>` with the hotfixed container image.
* Replace `<container_tag>` with the tag for the hotfixed container image.
* Replace `<updated_container_registry>` with a container registry to serve the hotfixed container image. You can use the OpenShift internal container image registry.
* Replace `<updated_container_project>` with a container project to use for the hotfixed container image.
```

> 

Reviewer 3: it didn't render as bullet points:
![image](https://github.com/openstack-k8s-operators/dataplane-operator/assets/17153022/13ec0c85-b7fc-4156-977c-f55a5b997014)
https://logserver.rdoproject.org/01/701/934bc9f064dc2630319cb4fc6704eff5abf367dd/github-check/dataplane-operator-docs-preview/9371f30/docs_build/docs_build/dataplane/index-upstream.html#proc_hotfixing-the-data-plane-container-content-rpms-dataplane

Reviewer: I wonder if it is a + missing earlier - can you see if there is a + after the line 18 step instruction:

"From a RHEL workstation, server, or virtual machine, collect the hotfix RPMs into a new directory"
+
----

Author: this was resolved

> . Update the affected `OpenStackDataPlaneNodeSet` resources by customizing the container locations to the hotfixed container locations. See <<_customizing_container_image_locations>> on how to set the hotfixed container locations.

Reviewer: ```suggestion
. Update the affected `OpenStackDataPlaneNodeSet` resources by customizing the container locations to the hotfixed container locations. For more information about how to set the hotfixed container locations, see <<_customizing_container_image_locations>>.
```

> . Create a new `OpenStackDataPlaneDeployment` resource that deploys the affected `OpenStackDataPlaneNodeSet` resources. See <<proc_deploying-the-data-plane_{context}>> on how to create `OpenStackDataPlaneDeployment` resources.

Reviewer: ```suggestion
. Create a new `OpenStackDataPlaneDeployment` resource that deploys the affected `OpenStackDataPlaneNodeSet` resources. For more information about how to create `OpenStackDataPlaneDeployment` resources, see <<proc_deploying-the-data-plane_{context}>>.
```

> . Create a new `OpenStackDataPlaneDeployment` resource that deploys the affected `OpenStackDataPlaneNodeSet` resources. For more information about how to create `OpenStackDataPlaneDeployment` resources, see <<proc_deploying-the-data-plane_{context}>>.

Reviewer 4: Would it make more sense to update openstackdataplanenodeset instead? Hotfixed image may be gone away on next 'normal' openstackdataplanedeployment run.

Author: It does say to update OpenStackDataPlaneNodeSet

+
[NOTE]

> The list of services can be restricted for the `OpenStackDataPlaneDeployment` to just those affected by the hotfix by using the `servicesOverride` field. See <<_overriding_services_for_the_deployment>>.

Reviewer: ```suggestion
You can restrict the list of services for the `OpenStackDataPlaneDeployment` to only those affected by the hotfix by using the `servicesOverride` field. For more information, see <<_overriding_services_for_the_deployment>>.
```
